### PR TITLE
converts class instance_variables to hash

### PR DESCRIPTION
**Pull request by syounger-ut**
## Why

As this is behaving like a model, it would be nice to have model functionality, such as .attributes and .to_hash.

The `.to_hash` it specifically tailored towards working with Dry::Struct

---

## What

add two methods to the Money class: `attributes` and `to_hash`.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  ruby
+
+DEPENDENCIES
+  bundler (~> 1.16)
+  farmoney!
+  rake (~> 10.0)
+  rspec (~> 3.0)
+
+BUNDLED WITH
+   1.16.6
```
Comment (tomchipchase) [inline Gemfile.lock]: Don't commit the Gemfile.lock for gems

File: lib/farmoney/money.rb [hunk 2]
```diff
@@ -20,6 +20,18 @@ def to_s
       format("£%.1i.%.2i", pounds, pence)
     end
 
+    def to_hash
+      attributes
+    end
```
Comment (tomchipchase) [inline lib/farmoney/money.rb]: `alias` or `alias_method` here

File: lib/farmoney/money.rb [hunk 3]
```diff
@@ -20,6 +20,18 @@ def to_s
...
+
+    def attributes
+      instance_variables.map do |attribute|
+        name = attribute_name(attribute)
+        value = instance_variable_get(attribute)
+        [name, value]
+      end.to_h
```
Comment (tomchipchase) [inline lib/farmoney/money.rb]: instance_variables.each_with_object({}) do |attribute, hash|
        hash[attribute_name(attribute)] = instance_variable_get(attribute)
      end

**Review by tomchipchase — CHANGES_REQUESTED** [review]

File: lib/farmoney/money.rb [hunk 4]
```diff
@@ -20,6 +20,14 @@ def to_s
       format("£%.1i.%.2i", pounds, pence)
     end
 
+    def attributes
+      instance_variables.each.with_object({}) do |attribute, hash|
```
Comment (syounger-ut) [inline lib/farmoney/money.rb:24]: according to [THIS](https://apidock.com/rails/Enumerable/each_with_object), each_with_object is deprecated. Looks like it is a result of [THIS](https://bugs.ruby-lang.org/issues/10549) request 4 years ago. Backed up by [THIS](https://stackoverflow.com/questions/14671881/how-does-enumwith-object-differ-from-enumeach-with-object/14672305#14672305) SO post explaining that .with_object just takes an Enumerable, and should not have such a narrow scope of just `each`.

**Pull request merged**